libs/s25main/world/MapBase.cpp

```diff
@@ -100,6 +100,21 @@ MapPoint MapBase::GetNeighbour2(const MapPoint pt, unsigned dir) const
     return MakeMapPoint(::GetNeighbour2(Position(pt), dir));
 }
 
+std::vector<MapPoint> MapBase::GetNeighbours(const MapPoint& pt) const
+{
+    unsigned yplus1 = pt.y == size_.y - 1 ? 0 : pt.y + 1;
```

Comment:
This looks wrong. What if pt.y+1==MapSize? Why not use the GetNeighbour?

Reply:
I used `GetNeighbour` before but the performance is much worse for larger map sizes.
This specific case should work, for example:
Assuming `size_.y = 16` and `pt.y = 15`. Then `pt.y + 1 = 16 (= size_.y)`. Therefore `(pt.y + 1) % size.y = 0`. `pt.y` cannot be `16` assuming `MapPoint` is always within the range `[0, size.y - 1]`. Or maybe I didn't get the point?